### PR TITLE
hw2_routing

### PR DESCRIPTION
**Pull request by Fenix-125**
add routing and some site structure updates

### CODE REVIEW

**Comment by Fenix-125** [comment]
did not see that package-lock.json is so large

**Comment by pavlo-berezin** [comment]
Please fix sidebar height :)
image

File: public/holder.js [hunk 1]
```diff
@@ -0,0 +1,974 @@
+/*!
```
Comment (pavlo-berezin) [inline public/holder.js:1]: What's this?

File: src/components/AboutUs.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            <Card className="text-center">
+                <Card.Header> </Card.Header>
+                <Card.Body>
+                    <Card.Title>WebDevelopment module project</Card.Title>
+                    <Card.Text>
+                        Author: Yuriy Pasichnyk<br/>
+                        Group: Computer Science 2<br/>
+                        Year: 2020<br/>
+                    </Card.Text>
+                    <Button target="_blank" href={"https://github.com/Fenix-125/mem-mmi-site"} variant="primary">Go to
```
Comment (pavlo-berezin) [inline src/components/AboutUs.js:18]: Please fix indentation, to something like this
```
<Button ...>
  Go to Github
</Button>
```

File: src/components/Catalog.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import Dash from "./Dash";
+import Card from "react-bootstrap/Card";
+import MemeCard from "./MemeCard";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+
+export default class Catalog extends React.Component {
+    render_cols(num) {
```
Comment (pavlo-berezin) [inline src/components/Catalog.js:10]: please use camelcase

File: src/components/Catalog.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        let res = [];
+        for (let i = 0; i < num; ++i) {
+            res.push(<Col><MemeCard/></Col>);
+        }
+        return res;
+    }
+
+    render_rows(num) {
```
Comment (pavlo-berezin) [inline src/components/Catalog.js:18]: This function kinda looks weird, consider refactoring it.

File: src/components/Dash.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, {Component} from "react";
+import {Col, Container, Row} from "react-bootstrap";
+import LeftBar from "./LeftBar";
+import "../styles/Dash.css"
+
+export default class Dash extends Component {
+    constructor(props) {
```
Comment (pavlo-berezin) [inline src/components/Dash.js:7]: no need of adding constructor here.

File: src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import Navbar from "react-bootstrap/Navbar";
+import Nav from "react-bootstrap/Nav";
+
+
+export default class Header extends React.Component {
+    render() {
+        return <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
```
Comment (pavlo-berezin) [inline src/components/Header.js:8]: Suggestion: to make it look better you can do it like this:
```
return (
  <Navbar ...>
  ...
)
```

File: src/components/Home.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import Carousel from "react-bootstrap/Carousel";
+import Jumbotron from "react-bootstrap/Jumbotron";
+
+export default class Home extends React.Component {
+    render() {
+        return (
+            <div className="bg-primary">
+                <Carousel>
+
+                    <Carousel.Item>
```
Comment (pavlo-berezin) [inline src/components/Home.js:11]: Put the data inside of some array, and just iterate it to generate markup for Carousel items.

File: src/components/MemeCard.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import Card from "react-bootstrap/Card";
+import ListGroupItem from "react-bootstrap/ListGroupItem";
+import ListGroup from "react-bootstrap/ListGroup";
+
+
+export default class MemeCard extends React.Component {
+    render() {
+        return (
+            <Card style={{width: '15rem', margin: '1%', gridRowGap: "1%"}}>
```
Comment (pavlo-berezin) [inline src/components/MemeCard.js:10]: Please move styles to css.

File: src/components/PageBar.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import Pagination from "react-bootstrap/Pagination";
+
+
+export default class MemeCard extends React.Component {
+    render() {
+        return (
+            <Pagination>
```
Comment (pavlo-berezin) [inline src/components/PageBar.js:8]: Consider not doing it so hardcoded.

File: src/styles/Dash.css [hunk 10]
```diff
@@ -0,0 +1,31 @@
+.sidebar {
+    position: fixed;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    min-height: 100vh !important;
```
Comment (pavlo-berezin) [inline src/styles/Dash.css:6]: please don't use importants, its bad for performance.

File: public/holder.js (hunk 1 above)
Comment (Fenix-125) [reply public/holder.js:1]: I should have been deleted it.

**Comment by Fenix-125** [comment]
> Please fix sidebar height :)
> image

**Pull request merged**